### PR TITLE
[dagster-aws] [docs] add docs for PipesEMRServerlessClient

### PR DESCRIPTION
**Pull request by danielgafni**
## Changelog

Insert changelog entry or "NOCHANGELOG" here.

- [x] `DOCS` _(Added docs for `PipesEMRServerlessClient`)_

### CODE REVIEW

**Comment by danielgafni** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24319?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#24319** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24319?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#24318** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24318?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @danielgafni and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/content/concepts/dagster-pipes/aws-emr-serverless.mdx [hunk 1]
```diff
@@ -0,0 +1,158 @@
+---
+title: "Integrating AWS EMR Serverless with Dagster Pipes | Dagster Docs"
+description: "Learn to integrate Dagster Pipes with AWS EMR Serverless to launch external code from Dagster assets."
+---
+
+# AWS EMR Serverless & Dagster Pipes
+
+This tutorial gives a short overview on how to use [Dagster Pipes](/concepts/dagster-pipes) with [AWS EMR Serverless](https://aws.amazon.com/emr-serverless/).
+
+The [dagster-aws](/\_apidocs/libraries/dagster-aws) integration library provides the <PyObject object="PipesEMRServerlessClient" module="dagster_aws.pipes" /> resource which can be used to launch AWS EMR Serverless jobs from Dagster assets and ops. Dagster can receive regular events like logs, asset checks, or asset materializations from jobs launched with this client. Using it requires minimal code changes on the job side.
```
Comment (PedramNavid) [inline docs/content/concepts/dagster-pipes/aws-emr-serverless.mdx:10]:
```suggestion
The [dagster-aws](/\_apidocs/libraries/dagster-aws) integration library provides the <PyObject object="PipesEMRServerlessClient" module="dagster_aws.pipes" /> resource, which can be used to launch AWS EMR Serverless jobs from Dagster assets and ops. Dagster can receive regular events such as logs, asset checks, or asset materializations from jobs launched with this client. Using it requires minimal code changes to your EMR jobs.
```

File: docs/content/concepts/dagster-pipes/aws-emr-serverless.mdx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+```Dockerfile
+# start from EMR image
+FROM public.ecr.aws/emr-serverless/spark/emr-7.2.0:latest
+
+USER root
+
+RUN python -m pip install dagster-pipes
+
+# copy the jon script
```
Comment (PedramNavid) [inline docs/content/concepts/dagster-pipes/aws-emr-serverless.mdx:57]: job?
```suggestion
# copy the job script
```

File: docs/content/concepts/dagster-pipes/aws-emr-serverless.mdx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+COPY . .
+
+USER hadoop
+```
+
+---
+
+## Step 2: Add dagster-pipes to the EMR Serverless job script
+
+Call `open_dagster_pipes` in the EMR Serverless script to create a context that can be used to send messages to Dagster:
```
Comment (PedramNavid) [inline docs/content/concepts/dagster-pipes/aws-emr-serverless.mdx:67]: is there a hello world for EMR Serverless scripts that we can use here too? 
maybe adding
```
from pyspark.sql import SparkSession
from dagster_pipes import (
    PipesEnvVarParamsLoader,
    PipesS3ContextLoader,
    open_dagster_pipes,
)

def main():
    with open_dagster_pipes() as pipes:
        spark = SparkSession.builder.appName("HelloWorld").getOrCreate()
        pipes.log.info("Hello from AWS EMR Serverless!")
        pipes.report_asset_materialization(
            metadata={"some_metric": {"raw_value": 0, "type": "int"}},
            data_version="alpha",
        )
        spark.stop()


if __name__ == "__main__":
    main()
```

so it's clear this is running in EMR and it gives people an idea of how it fits in with spark?

**Review by PedramNavid — APPROVED** [review]
Just a few small comments, but overall this looks good to me.

File: docs/content/concepts/dagster-pipes/aws-emr-serverless.mdx (hunk 3 above)
Comment (danielgafni) [reply docs/content/concepts/dagster-pipes/aws-emr-serverless.mdx:67]: Added some actual Spark usage

**Comment by danielgafni** [comment]
CI failing due to broken `master`

**Pull request merged**